### ISSUE TITLE
Node Repair: Expand to Handle More Node Statuses

### ISSUE THREAD

**Issue by ahmadbabaeimoghadam** [aws/karpenter-provider-aws#7479]
### Description

**What problem are you trying to solve?**

The Node Repair feature introduced in Karpenter v1.1.0 currently only recycles nodes in the `NotReady` state ([reference](https://github.com/aws/karpenter-provider-aws/blob/v1.1.0/pkg/cloudprovider/cloudprovider.go#L254)).

In our environment, nodes occasionally enter an `Unknown` state with the following condition:

```yaml
- type: Ready
  status: Unknown
  reason: NodeStatusUnknown
  message: Kubelet stopped posting node status.
```

It would be extremely helpful if the Node Repair feature could also handle nodes in the Unknown state, in addition to NotReady.

**How important is this feature to you?**

This issue arises frequently in our environment, significantly impacting cluster reliability. Extending Node Repair to address this case would greatly improve our operations.

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
> In our environment, nodes occasionally enter an Unknown state with the following condition:

Do you know the reason why nodes enter this state in the first place? Does the kubelet just die?

**Comment by ahmadbabaeimoghadam** [comment]
> Do you know the reason why nodes enter this state in the first place? Does the kubelet just die?

Yes, the reason is memory pressure. Initially, the node enters the MemoryPressure condition, and after some time, it transitions to an Unknown state because the Kubelet crashes due to insufficient memory. We will address this by fine-tuning the memory requests and limits to prevent such situations, but having this feature as a node repair policy would be very helpful.
